src/Files.App/BaseLayout.cs

```diff
@@ -587,14 +596,23 @@ private async Task LoadMenuItemsAsync()
 			ItemContextMenuFlyout.PrimaryCommands.Clear();
 			ItemContextMenuFlyout.SecondaryCommands.Clear();
 			var (primaryElements, secondaryElements) = ItemModelListToContextFlyoutHelper.GetAppBarItemsFromModel(items);
-			primaryElements.OfType<AppBarButton>().ForEach(i =>
-			{
-				i.Click += new RoutedEventHandler((s, e) => ItemContextMenuFlyout.Hide()); // Workaround for WinUI (#5508)
-			});
 			primaryElements.ForEach(i => ItemContextMenuFlyout.PrimaryCommands.Add(i));
 			secondaryElements.OfType<FrameworkElement>().ForEach(i => i.MinWidth = Constants.UI.ContextMenuItemsMaxWidth); // Set menu min width
 			secondaryElements.ForEach(i => ItemContextMenuFlyout.SecondaryCommands.Add(i));
 
+			// Workaround for WinUI (#5508)
+			var closeHandler = new RoutedEventHandler((s, e) => ItemContextMenuFlyout.Hide());
+			primaryElements
+				.OfType<AppBarButton>()
+				.ForEach(button => button.Click += closeHandler);
+			secondaryElements
+				.OfType<AppBarButton>()
+				.Select(item => item.Flyout)
+				.OfType<MenuFlyout>()
+				.SelectMany(menu => menu.Items)
+				.OfType<MenuFlyoutItem>()
+				.ForEach(button => button.Click += closeHandler);
```

Comment:
Those lines are a repetition of l.541 to l.553.
Would it be possible to fuse them inside a method?

Reply:
It's done.